### PR TITLE
Feature/implement room and retrofit/arc 675

### PR DESCRIPTION
**Pull request by ted3x**

### CODE REVIEW

**Review by DavitSoitashviliSpace — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/c0d3in3/movieapp/data/remote/ApiService.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import retrofit2.http.Path
+import retrofit2.http.Query
+
+interface ApiService {
+    @GET("movie/{movie_id}")
+    fun getMovieDetails(@Path("movie_id") movieId: Int) : Call<Movie>
+
+    @GET("movie/top_rated")
+    fun getTopRatedMovies(@Query("page") page: Int = 1) : Call<MovieCollection>
+
```
[the remark this answers is not in the document]
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/data/remote/ApiService.kt:17]: I think it's better to have these two functions splitted, than having another parameter to divide them.

File: app/src/main/java/com/c0d3in3/movieapp/data/remote/ApiService.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import com.c0d3in3.movieapp.models.network.Results
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Path
+import retrofit2.http.Query
+
+interface ApiService {
+    @GET("movie/{movie_id}")
+    fun getMovieDetails(@Path("movie_id") movieId: Int) : Call<Movie>
+
```
[the remark this answers is not in the document]
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/data/remote/ApiService.kt:14]: Changed.

File: app/src/main/java/com/c0d3in3/movieapp/models/network/Movie.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    @SerializedName("belongs_to_collection") val belongsToCollection : Collection,
+    val budget : Int,
+    val genres : List<Genres>,
+    val homepage : String,
+    val id : Int,
+    @SerializedName("imdb_id") val imdbId : String,
+    @SerializedName("original_language") val originalLanguage : String,
+    @SerializedName("original_title") val originalTitle : String,
+    val overview : String,
+    val popularity : Double,
```
[the remark this answers is not in the document]
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/models/network/Movie.kt:19]: Added nullable types.

File: app/src/main/java/com/c0d3in3/movieapp/data/remote/RetrofitClient.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        .build()
+
+    private val retrofit = Retrofit.Builder()
+        .baseUrl("https://api.themoviedb.org/3/")
+        .client(okHttpClient)
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+
+    var service: ApiService = retrofit.create(ApiService::class.java)
+}
```
[the remark this answers is not in the document]
Comment (ted3x) [reply app/src/main/java/com/c0d3in3/movieapp/data/remote/RetrofitClient.kt:27]: Changed

**Review by Edna001 — APPROVED** [review]

**Review by DavitSoitashviliSpace — APPROVED** [review]

**Pull request merged**